sdk/core/azure-core/inc/http/http.hpp

```diff
@@ -215,30 +220,50 @@ namespace Azure { namespace Core { namespace Http {
       return std::string(url.begin(), firstPosition);
     }
 
+    std::string GetQueryString() const;
+
     Request(
         HttpMethod httpMethod,
         std::string const& url,
         BodyStream* bodyStream,
-        std::vector<uint8_t> bodyBuffer)
-        : _method(std::move(httpMethod)), _url(parseUrl(url)), m_bodyStream(bodyStream),
-          m_bodyBuffer(std::move(bodyBuffer)), m_retryModeEnabled(false)
+        std::vector<uint8_t> bodyBuffer,
+        BodyType responseBodyType)
+        : m_method(std::move(httpMethod)), m_url(saveAndRemoveQueryParameter(url)),
+          m_bodyStream(bodyStream), m_bodyBuffer(std::move(bodyBuffer)), m_retryModeEnabled(false),
+          m_bodyTypeForResponse(std::move(responseBodyType))
     {
-      // TODO: parse url
     }
 
   public:
     Request(HttpMethod httpMethod, std::string const& url)
-        : Request(httpMethod, url, BodyStream::null, std::vector<uint8_t>())
+        : Request(httpMethod, url, BodyType::Buffer)
     {
     }
 
+    Request(HttpMethod httpMethod, std::string const& url, BodyType respondBodyType)
+        : Request(httpMethod, url, BodyStream::null, std::vector<uint8_t>(), respondBodyType)
+    {
+    }
+
+    // Default request with buffer will return buffer response
     Request(HttpMethod httpMethod, std::string const& url, std::vector<uint8_t> bodyBuffer)
-        : Request(httpMethod, url, BodyStream::null, std::move(bodyBuffer))
+        : Request(httpMethod, url, std::move(bodyBuffer), BodyType::Buffer)
```

Comment:
Does it mean, if the request body is a buffer, then the response body must be a buffer. If the request body is a stream, then the response body must be a stream? Why?

Reply:
What if somebody wants to use buffer as request body and stream as response body, or the other way around?

Reply:
not really. We have another more general constructor to set up the bodyType. But it is only possible for a request with bodyBuffer.

So, what you mentioned is the default behavior. 

If users create a Request with bodyBuffer without passing a bodyType, it would produce Response with bodyBuffer.

Users can set the value of bodyType to `stream` so the request will produce a Response with body Stream. 

When a request contains a bodyStream, it can only produce a response with bodyStream. This is by design.

Reply:
okay